### DISCUSSION TITLE
Infectivity and symptomatic states

### DISCUSSION THREAD

**Discussion by jubicker** [SciCompMod/memilio#731] [Ideas]
As there is not necessarily a correlation betweeen infectivity and symptomatic state and this correlation can differ for various diseases, we would like to make (symptomatic) state transitions and calculation of infectivity more flexible.

Proposed approach:
Make the infection states only symptomatic states and provide an interface that handles the calculation of infectivity and the state transitions. With this interface, the calculation of infectivity and the state transitions a subject to user input, but we provide some default implementations for that.

Open issues:

1)  What symptomatic states do we want to have?
- no symptoms, mild symptoms, severe symptoms (hospitalized?), critical symptoms (icu?) 
2) What default implementations should we provide for the interface?
- Option 1: State transitions and infectivity are completely decoupled. State transitions follow are fixed distribution (exponential or discrete?) with given rates and infectivity is calculated using an individual invlogit func for every person.
- Option 2: Infectivity is dependent on the symptomatic state. State transitions are e.g. exponentially distributed with given rates and infectivity is the rate of the symptomatic state a person is in.

**Comment by jubicker** [comment]
@xsaschako @mknaranja @DavidKerkmann @khoanguyen-dev feel free to share your thoughts and suggestions here

**Comment by xsaschako** [comment]
Just to add to this discussion (if there is still any):
I'll link our preliminary discussion here : https://github.com/SciCompMod/memilio/issues/688

@DavidKerkmann  and I talked about this. We think the following solution should be good:

We do the following symptomatic states:

NonSymptomatic
Symptomatic
Severe
Critical
Dead

Furthermore there is a viral load and a parameter if the User wants to couple it to the Symptomatic State (where we need to research)

We need to pay attention to Reinfections. (Waning) Immunity would be the candiate for this.
